### PR TITLE
Read variables from .env file in project dir if it is provided.

### PR DESCRIPTION
**Pull request by julien-sebire**
Reads variables from .env file in project dir if it is provided. If no `.env` file is provided, just does nothing.
This allows us to then access variables values in `$_ENV` global array.
Works for CLI, Apache, Nginx, ...

Example:

.env file:
```
GENERIS_CACHE_URI=http://my-redis.com/blah/blah/blah
```

.php file:
```
echo $_ENV['GENERIS_CACHE_URI']; // => http://my-redis.com/blah/blah/blah
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -56,7 +56,8 @@
         "pimple/pimple": "^3.0",
         "imsglobal/lti": "^3.0",
         "paragonie/random_compat": "^2.0",
-        "oat-sa/composer-npm-bridge": "^0.3.0"
+        "oat-sa/composer-npm-bridge": "^0.3.0",
+        "symfony/dotenv": "^5.0@dev"
```
Comment (Babacooll) [inline composer.json]: Why do we need ^5.0 version ?

Comment (julien-sebire) [reply composer.json]: That's right, I just picked the version that popped out when `composer requiring` the repo... Fixed to the last stable version.

File: composer.json [hunk 2]
```diff
@@ -56,7 +56,8 @@
         "pimple/pimple": "^3.0",
         "imsglobal/lti": "^3.0",
         "paragonie/random_compat": "^2.0",
-        "oat-sa/composer-npm-bridge": "^0.3.0"
+        "oat-sa/composer-npm-bridge": "^0.3.0",
+        "symfony/dotenv": "^4.0"
```
Comment (Babacooll) [inline composer.json]: Shouldn't it be ^4.3 ?

File: models/classes/mvc/Bootstrap.php [hunk 3]
```diff
@@ -89,6 +90,13 @@ class Bootstrap implements ServiceManagerAwareInterface
      */
     public function __construct($configuration)
     {
+        $envDir = __DIR__ . '/../../../../';
+        $envFile = $envDir. '.env';
+        if (file_exists($envFile)) {
+            $dotenv = new Dotenv();
+            $dotenv->load($envFile);
```
Comment (peetya) [inline models/classes/mvc/Bootstrap.php]: What do you think about using `loadEnv` instead of `load`?

Comment (julien-sebire) [reply models/classes/mvc/Bootstrap.php]: You're right. Fixed.

File: composer.json (hunk 2 above)
Comment (julien-sebire) [reply composer.json]: I even went to `~4.3` so that we could include higher 4.x version also.

File: models/classes/mvc/Bootstrap.php [hunk 4]
```diff
@@ -89,6 +90,13 @@ class Bootstrap implements ServiceManagerAwareInterface
      */
     public function __construct($configuration)
     {
+        $envDir = __DIR__ . '/../../../../';
+        $envFile = $envDir. '.env';
```
Comment (ekkinox) [inline models/classes/mvc/Bootstrap.php]: - would merge in one variable
- would use `DIRECTORY_SEPARATOR` to avoid windows issues

Comment (julien-sebire) [reply models/classes/mvc/Bootstrap.php]: Fixed in PR https://github.com/oat-sa/tao-core/pull/2238.

**Pull request merged**